product_pricelist_brand/models/product_pricelist.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2020 NextERP Romania SRL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    @api.multi
+    def _compute_price_rule(self, products_qty_partner, date=False,
```

Comment:
This looks like a lot of duplicated code to me.
The base code is definitely not meant to be extended. In my version I override `_compute_price_rule_get_items`, I don't really like it neither but still I think it is a bit cleaner.
Unless there is a good reason to  work on `_compute_price_rule` that I am missing ?